processing/src/main/java/org/apache/druid/query/aggregation/hyperloglog/HyperUniquesAggregatorFactory.java

```diff
@@ -137,17 +137,21 @@ public VectorAggregator factorizeVector(final VectorColumnSelectorFactory select
 
   /**
    * Validates whether the aggregator supports the input column type.
+   * Supported column types are complex types of hyperUnique, preComputedHyperUnique, as well as UNKNOWN_COMPLEX.
```

Comment:
I wanted to check if this change is fine with primitive types - earlier in case of primitive selectors, we'd have returned `NoopVectorAggregator.instance`. Now it will throw an error.

Reply:
Also, the check on Line 131 in this file seems redundant after we have validated that the type is a supported type and complex.

Reply:
My bad, makes sense. I've moved the validateInputs() check to a later point, such that the NoopXyz.instance() are returned like previously. Have also added a basic test to validate the NoopXyz.instance() flows for factorize methods of this class.